lib/swagger/schema.rb

```diff
@@ -42,23 +41,29 @@ def resolve_all_refs(schema)
     end
 
     def resolve_refs
-      items_and_props = [deep_select('items'), deep_select('properties')].flatten.compact
-      items_and_props.each do |item|
-        key = item.delete('$ref')
-        next if remote_ref? key
-        model = root.definitions[key]
-        item.merge!(model)
+      items_and_props = [deep_select('items'), deep_select('properties').map(&:values)].flatten.compact
```

Comment:
I really don't think it matters in this case (I had to run my benchmark a million times to get results over noise) but `flatten` (and even `flatten!`) is an expensive operation compared to array merging.